golem-cli/tests/worker.rs

```diff
@@ -724,3 +725,59 @@ fn worker_list(
 
     Ok(())
 }
+
+fn worker_update(
+    (deps, name, cli): (
+        Arc<dyn TestDependencies + Send + Sync + 'static>,
+        String,
+        CliLive,
+    ),
+) -> Result<(), Failed> {
+    let cfg = &cli.config;
+    let component_id = make_component(deps, &format!("{name} worker_update"), &cli)?.component_id;
+    let worker_name = format!("{name}_worker_update");
+
+    let workers_list = || -> Result<WorkersMetadataResponse, Failed> {
+        cli.run(&[
+            "worker",
+            "list",
+            &cfg.arg('C', "component-id"),
+            &component_id,
+            &cfg.arg('f', "filter"),
+            format!("name like {}_worker", name).as_str(),
+        ])
+    };
+
+    let _: WorkerId = cli.run(&[
+        "worker",
+        "add",
+        &cfg.arg('w', "worker-name"),
+        &worker_name,
+        &cfg.arg('C', "component-id"),
+        &component_id,
+    ])?;
+    let original_updates = workers_list()?.workers[0].updates.len();
+
+    cli.run_unit(&[
+        "worker",
+        "update",
+        &cfg.arg('w', "worker-name"),
+        &worker_name,
+        &cfg.arg('C', "component-id"),
+        &component_id,
+        &cfg.arg('m', "mode"),
+        "auto",
+        &cfg.arg('t', "target-version"),
+        "1",
+    ])?;
+    let worker_updates_after_update = workers_list()?.workers[0].updates[0].clone();
+    let target_version = if let UpdateRecord::PendingUpdate(pu) = worker_updates_after_update {
```

Comment:
Could you please match `SuccessfulUpdate` as well? In case of slow test runner the server might be able to update worker faster than the test asks for worker list.

Reply:
Or even better you can put `workers_list()?.workers[0].component_version` into `else` branch.